qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AddRoleToUserSteps.java

```diff
@@ -0,0 +1,154 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Map.Entry;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.contrib.qaframework.page.AddEditUserPage;
+import org.openmrs.contrib.qaframework.page.AdministrationPage;
+import org.openmrs.contrib.qaframework.page.HomePage;
+import org.openmrs.contrib.qaframework.page.ManageUserPage;
+
+public class AddRoleToUserSteps extends Steps {
+	
+	private ManageUserPage manageUserPage;
+	private AddEditUserPage addEditUserPage;
+	private AdministrationPage administrationPage;
+        private static final String NURSE_PASSWORD = "Nurse321";
+        private static final String NURSE_USERNAME = "newNurse" + new Random().nextInt(1024);
+    
+    private void reLoginAsUser() throws InterruptedException {
+        goToLoginPage().login(NURSE_USERNAME, NURSE_PASSWORD);
+    }
+
+    private void reLoginAsAdmin() throws InterruptedException {
+        goToLoginPage().loginAsAdmin();
+    }
+    
+    private void fillInRoleModules(Map<String, Integer> roleModules) {
+        roleModules.put("roleStrings.Anonymous", 0);
+        roleModules.put("roleStrings.Authenticated", 0);
+        roleModules.put("roleStrings.Organizational:SystemAdministrator", 2);
+        roleModules.put("roleStrings.SystemDeveloper", 9);
+        roleModules.put("roleStrings.Application:AdministersSystem", 1);
+        roleModules.put("roleStrings.Organizational:Doctor", 3);
+        roleModules.put("roleStrings.Organizational:Nurse", 4);
+        roleModules.put("roleStrings.Organizational:RegistrationClerk", 3);
+    }
+	
+	@Before(RunTest.HOOK.SELENIUM_ADD_ROLE_TO_USER)
+	public void visitHomePage() {
+		initiateWithLogin();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_ADD_ROLE_TO_USER)
+	public void destroy() throws InterruptedException {
+		quit();
+	}
+	
+	@Given("user clicks on system administration app")
+	public void clickOnSystemAdministrationApp() {
+		manageUserPage = homePage.goToAdministration().clickOnManageUsers();
+	}
+	
+	@Then("system loads manage users page")
+	public void systemLoadsManageUsersPage() {
+		assertTrue(textExists("User Management"));
+	}
+	
+	@When("user fills in person details")
+	public void fillInPersonDetails() {
+                manageUserPage.clickOnAddUser().createNewPerson().fillInPersonName("Super", "Nurse", NURSE_USERNAME, NURSE_PASSWORD);
+	}
+	
+	@When("user clicks the add user link")
+	public void clickOnAddUserLink() {
+                addEditUserPage = manageUserPage.clickOnAddUser();
+                addEditUserPage.createNewPerson();
+	}
+	
+	@And("user assigns roles to the created user")
+	public void assignRolesToUser() throws InterruptedException {
+        Map<String, Integer> roleModules = new HashMap<>();
```

Comment:
Please go through all the files and fix the indention.

Reply:
alirght, lemi do that